widget/table.go

```diff
@@ -512,7 +512,13 @@ func (c *tableCells) Resize(s fyne.Size) {
 	if s == c.size {
 		return
 	}
+	cSizeBeforeResize := c.size
 	c.BaseWidget.Resize(s)
+	if c.size.Height > cSizeBeforeResize.Height {
+		// table height has increased.
+		// call ScrollToTop() otherwise empty rows will appear at the top of the table.
+		c.t.ScrollToTop()
```

Comment:
I see that this fixes some instances of the issue, but if there are 1000 items and you are looking at 900 and the view gets larger then scrolling to the top won't be the right thing to do

Reply:
I noticed that the if condition becomes true only when the window is resized to a size big enough that there are no hidden items. So `ScrollToTop()` doesn't cause any problems.
But there should be a better fix. In my comment below I describe what I noticed about the value of `c.size.Height`.

Reply:
Your explanation makes sense - but it does not match what the code is doing now, so it will still scroll up some times when it is not desired.
I agree there is likely a better fix, have you any ideas?

Reply:
I agree there is likely a better fix, have you any ideas?
here is an idea I tried at first: https://github.com/fyne-io/fyne/commit/cfaeb96b6476d2d254809e19858f4e62cb736b94
I deleted it by force pushing because I noticed it doesn't work when I make the window smaller. but if I can fix it further it might be a better solution.